PerfectWeddingWeb/src/main/java/com/infoshareacademy/services/ServiceProviderService.java

```diff
@@ -28,16 +31,40 @@ public ServiceProviderService(ServiceProviderRepo serviceProviderRepo, ServicePr
         this.serviceProviderMapper = serviceProviderMapper;
     }
 
-
     public List<ServiceProvider> returnAllServiceProviders() {
         return serviceProviderRepo.getServiceProvidersList();
     }
 
-    public List<ServiceProvider> findById(Integer id) {
+    public ServiceProvider deActivate(Integer id) {
+        serviceProviderRepo.getServiceProvidersList().stream()
+                .filter(serviceProvider -> serviceProvider.getCurrentID() == id)
+                .forEach(serviceProvider -> serviceProvider.setActive(false));
+        return null;
+
+    }
+
+    public ServiceProvider editById(Integer id) {
         return serviceProviderRepo.getServiceProvidersList()
                 .stream()
                 .filter(serviceProvider -> serviceProvider.getCurrentID() == id)
-                .collect(Collectors.toList());
+                .findFirst()
+                .orElseThrow();
+    }
+
+    public void editProvider(ServiceEditProviderDto serviceEditProviderDto) throws IOException {
+        ServiceProvider serviceProvider = editById(serviceEditProviderDto.getId());
+        serviceProvider.setCompanyName(serviceEditProviderDto.getCompanyName());
+        serviceProvider.setOwnerName(serviceEditProviderDto.getOwnerName());
+        serviceProvider.setOwnerSurname(serviceEditProviderDto.getOwnerSurname());
```

Comment:
przez to, że nadpisujemy tutaj np. name i surname, a nie było ich w serviceEditProviderDto (bo nie ma w maperze), to czyścimy te wartości (przypisujemy null) 😱 
Trzeba dopasować dto-mapper-encja, żeby podczas edycji przypadkiem nie usuwać wartości 😉 
Przed edycją:
![Screenshot 2022-01-29 at 21 07 51](https://user-images.githubusercontent.com/25279453/151676045-44e905d3-1722-4566-91ee-fe3343be3029.png)
Po edycji:
![Screenshot 2022-01-29 at 21 08 50](https://user-images.githubusercontent.com/25279453/151676051-2af025c1-ca8c-448a-8f9f-b81f361dba17.png)